packages/jupyter-ai/jupyter_ai/handlers.py

```diff
@@ -289,13 +289,17 @@ def filter_predicate(local_model_id: str):
             else:
                 return model_id in self.allowed_models
 
+        def filter_model_list(model_list: Optional[List[str]]) -> List[str]:
+            if model_list is not None:
+                return list(filter(filter_predicate, provider.models))
+            else:
+                return []
+
         # filter out every model w/ model ID according to allow/blocklist
         for provider in providers:
-            provider.models = list(filter(filter_predicate, provider.models))
-            provider.chat_models = list(filter(filter_predicate, provider.chat_models))
-            provider.completion_models = list(
-                filter(filter_predicate, provider.completion_models)
-            )
+            provider.models = filter_model_list(provider.models)
+            provider.chat_models = filter_model_list(provider.chat_models)
+            provider.completion_models = filter_model_list(provider.completion_models)
```

Comment:
Instead of defining `filter_model_list()` to handle the case of a `provider.{models,chat_models,completion_models}` being `None`, it would be easier to use the `or` operator like so:

```py
provider.models = list(filter(filter_predicate, provider.models or []))
...
```

This way, whenever `provider.models is None`, the expression in the second argument to `filter()` always evaluates to a list.

Reply:
Also note that the definition of `filter_model_list()` should return

```py
return list(filter(filter_predicate, model_list))
```

instead of 

```py
return list(filter(filter_predicate, provider.models))
```